clients/vault/src/system.rs

```diff
@@ -431,11 +431,18 @@ impl VaultService {
 			oracle_agent,
 			self.config.payment_margin_minutes,
 		);
+
+		let shutdown_clone = self.shutdown.clone();
 		service::spawn_cancelable(self.shutdown.subscribe(), async move {
-			// TODO: kill task on shutdown signal to prevent double payment
```

Comment:
I followed how [Interlay]([Interlay](https://github.com/pendulum-chain/spacewalk/pull/486/commits/80fba1217cb81e368ee6815d3f75516d87c214e1)) did it.

Reply:
here: https://github.com/interlay/interbtc-clients/commit/622f36b5efd538c2011ecfebce2425de6be17388